docs/src/extend/log-management.md

```diff
@@ -0,0 +1,210 @@
+---
+
+This document describes how to extend the log management capabilities of %%te%% using custom log plugins.
+By default, %%te%% supports retrieving file-based logs.
+But now, it can collect logs from other sources like systemd journal, Docker containers
+or any custom log sources using plugins.
+
+## Overview
+
+The %%te%% agent supports extensible log management through a plugin system:
+
+* The built-in `file` plugin handles traditional file-based logs (the default behavior)
+* Additional plugins can be installed to support other log sources
+* Each plugin can provide multiple log types
+* Plugins are discovered and executed automatically by `tedge-agent`
+
+## How Log Plugins Work
+
+A log plugin is an executable that implements a simple command-line interface with the following requirements:
+
+* Implement two sub-commands:
+  1. **`list`** - Returns all log types supported by the plugin (one per line)
+  2. **`get <log-type> [--since <timestamp>] [--until <timestamp>]`** - Retrieves logs for the specified type within the given time range
+* Must be executable and installed in `/usr/share/tedge/log-plugins`.
+* Must exit with code 0 for successful `list` command (used to validate the plugin).
+* Should output logs to stdout for the `get` command.
+* Time filters `--since` and `--until` are passed as seconds since epoch.
+* Should handle errors gracefully and exit with non-zero codes on failure.
+
+The agent automatically:
+* Discovers plugins at startup by running their `list` command to gather their supported log types.
+* Publishes the supported log types to the device's `log_upload` command metadata topic (e.g: `te/device/main///cmd/log_upload`)
+  with a plugin suffix in the format `<log-type>::<plugin-name>` (e.g., `mosquitto::journald`)
+* Routes `log_upload` requests to the `get` command of the appropriate plugin based on the type suffix.
+* The `dateFrom` and `dateTo` parameters in the command are passed to the plugin as `--since` and `--until` arguments.
+* Further filtering by `searchText` and tail `lines` are done by the agent itself.
+* Detects any new plugin installations dynamically.
+* Refresh the supported log types by reloading the plugins when any new software is installed or configuration is updated.
+
+Log plugins are installed at `/usr/share/tedge/log-plugins`, by default.
+
+This plugin root directory can be changed using:
+```sh
+sudo tedge config set log.plugin_paths /usr/local/share/tedge/log-plugins,/usr/share/tedge/log-plugins
+```
+
+Multiple plugin directories can be specified if a layered directory structure is desired,
+where plugins from directories earlier in the list gets precedence over the latter ones.
+For example, with the above configuration, a `file` plugin in `/usr/local/share/tedge/log-plugins`
+would override the one with the same name in `/usr/share/tedge/log-plugins`.
+This mechanism can be used to override the factory plugins when needed.
+
```

Comment:
I would put here the section about `sudo` configuration.

Reply:
Resolved by fa5b834d6b